sandbox/modules/identity/src/internalClusterTest/java/org/opensearch/identity/BasicAuthenticationIT.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity;
+
+import org.opensearch.action.admin.cluster.node.info.NodeInfo;
+import org.opensearch.action.admin.cluster.state.ClusterStateResponse;
+import org.opensearch.client.Request;
+import org.opensearch.client.RequestOptions;
+import org.opensearch.client.Response;
+import org.opensearch.client.RestClient;
+import org.opensearch.common.io.stream.NamedWriteableRegistry;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.plugins.NetworkPlugin;
+import org.opensearch.plugins.Plugin;
+import org.opensearch.test.OpenSearchIntegTestCase;
+import org.opensearch.test.OpenSearchIntegTestCase.ClusterScope;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.Transport;
+import org.opensearch.transport.TransportInterceptor;
+import org.opensearch.transport.TransportMessageListener;
+import org.opensearch.transport.TransportRequest;
+import org.opensearch.transport.TransportRequestOptions;
+import org.opensearch.transport.TransportResponse;
+import org.opensearch.transport.TransportResponseHandler;
+import org.opensearch.transport.TransportService;
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertNoTimeout;
+
+@ClusterScope(scope = OpenSearchIntegTestCase.Scope.TEST, numDataNodes = 0)
+public class BasicAuthenticationIT extends HttpSmokeTestCaseWithIdentity {
+
+    public static Map<String, String> interceptedTokens = new HashMap<>();
```

Comment:
Resolve or add comments todos; this should be concurrency safe, it would be ideal if the test 'waited for N' tokens or some other identifying mechanism to filter.  Giving me strong [TestAuditlogImpl.java](https://github.com/opensearch-project/security/blob/main/src/test/java/org/opensearch/security/auditlog/integration/TestAuditlogImpl.java) vibes.

Might be better to move into its own class off it this directory.